pkg/storage/pebble.go

```diff
@@ -1863,7 +1863,7 @@ func pebbleExportToSst(
 			if paginated && (isNewKey || stopMidKey) && reachedTargetSize {
```

Comment:
I wonder if we should do `(isNewKey && reachedTargetSize) || (stopMidKey && reachedMaxSize)` and then move the if maxSize err return up, to an `else if` block on this `if` ? That'd mean we'd loop around before checking it, but the err return is already after we added to the writer, so I don't think it'd be a big deal to do it at the top of the next iteration instead?

Reply:
That's an interesting point, then we will keep adding till we reach next key after targetSize, but instead of failing we'll stop mid key. So we will only fail if single version exceeds max-target.
Is my understanding correct?